adam/experiment/run_pipeline.py

```diff
@@ -0,0 +1,646 @@
+"""
+A script intended to run the entire M3 pipeline.
+
+Note that this script does not take care of starting the object segmentation server. You have to do
+that separately.
+
+Note also that this writes to then reads from a params file for ADAM. If you run this again with the
+same params file name while a job is running, you will overwrite the old parameters! So be careful.
+"""
+import atexit
+from datetime import datetime
+from io import StringIO
+from locale import getpreferredencoding
+import logging
+import os
+from os import getenv
+from pathlib import Path
+import shlex
+from shutil import copytree
+from subprocess import run, CalledProcessError
+from typing import List, Mapping, NewType, Optional, Sequence
+
+import yaml
+
+from vistautils.parameters_only_entrypoint import parameters_only_entry_point
+from vistautils.parameters import Parameters, ParameterError
+
+logger = logging.getLogger(__name__)
+
+
+SlurmJobID = NewType("SlurmJobID", str)
+Email = NewType("Email", str)
+MailType = NewType("MailType", str)
+
+slurm_jobs_scheduled: List[SlurmJobID] = []
+
+
+def stop_slurm_jobs():
+    """
+    Exit handler to clean up Slurm jobs.
+    """
+    for job_id in slurm_jobs_scheduled:
+        logger.info("Canceling previously scheduled job %s...", job_id)
+        cancel_command = ["scancel", str(job_id)]
+        print(" ".join(shlex.quote(part) for part in cancel_command))
```

Comment:
Is this a debugging print statement or an intended one? It's unclear to me which its suppose to be.

Reply:
Actual one, if I recall correctly..